frontend/src/features/contacts/contact/create/Person/CreatePersonForm.tsx

```diff
@@ -0,0 +1,262 @@
+import { UnsavedChangesPrompt } from 'components/common/form/UnsavedChangesPrompt';
+import GenericModal from 'components/common/GenericModal';
+import { Stack } from 'components/common/Stack/Stack';
+import { CountryCodes } from 'constants/countryCodes';
+import { Address, ContactEmailList, ContactPhoneList } from 'features/contacts/contact/create';
+import { personCreateFormToApiPerson } from 'features/contacts/contactUtils';
+import {
+  Formik,
+  FormikHelpers,
+  FormikProps,
+  getIn,
+  validateYupSchema,
+  yupToFormErrors,
+} from 'formik';
+import { useApiAutocomplete } from 'hooks/pims-api/useApiAutocomplete';
+import { useApiContacts } from 'hooks/pims-api/useApiContacts';
+import { IAutocompletePrediction } from 'interfaces';
+import { defaultCreatePerson, ICreatePersonForm } from 'interfaces/ICreateContact';
+import { useMemo, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { AiOutlineExclamationCircle } from 'react-icons/ai';
+import { useHistory } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import useAddressHelpers from '../address/useAddressHelpers';
+import { PadBox } from '../styles';
+import CommentNotes from './comments/CommentNotes';
+import * as Styled from './styles';
+import { hasAddress, hasEmail, hasPhoneNumber, validationSchema } from './validation';
+
+export interface ICreatePersonFormProps {}
+
+/**
+ * Formik-connected form to Create Individual Contacts
+ */
+export const CreatePersonForm: React.FunctionComponent<ICreatePersonFormProps> = props => {
+  const history = useHistory();
+  const { postPerson } = useApiContacts();
+
+  // validation needs to be adjusted when country == OTHER
+  const { countries } = useAddressHelpers();
+  const otherCountryId = useMemo(
+    () => countries.find(c => c.code === CountryCodes.Other)?.value?.toString(),
+    [countries],
+  );
+
+  const onValidate = (values: ICreatePersonForm) => {
+    try {
+      validateYupSchema(values, validationSchema, true, { otherCountry: otherCountryId });
+      // combine yup schema validation with custom rules
+      const errors = {} as any;
+      if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
+        errors.needsContactMethod =
+          'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
+      }
+      return errors;
+    } catch (err) {
+      return yupToFormErrors(err);
+    }
+  };
+
+  const onSubmit = async (
+    values: ICreatePersonForm,
+    { resetForm, setSubmitting }: FormikHelpers<ICreatePersonForm>,
+  ) => {
+    try {
+      await postPerson(personCreateFormToApiPerson(values));
+      setSubmitting(false);
+      resetForm({ values });
+      toast.info('Contact added successfully');
+      history.push('/contact/list');
+    } catch (error) {
+      toast.error('Failed to add contact', { autoClose: 7000 });
+      history.push('/contact/list');
```

Comment:
Is this correct? Normally I'd expect a form that I'm working on to stay in context if a save error occurs. That way I have a chance to try to resubmit/correct an error without losing all of my work.

Of course, do whatever the requirements say, but I personally don't like this behaviour.

Reply:
I can remove the navigation. Requirements didn't specify what to do on error. What did you do for leases?

Reply:
Just display the error, no navigation. Allows them to attempt to submit again.

Reply:
okay - will remove